runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -59,6 +60,61 @@ class MemoryManager final {
     bool contains(void *ptr) { return _impl.contains(ptr); }
 };
 
+class SharedLibraryManager final {
```

Comment:
One more comment :smile: 

I know you were also looking into the performance concerns with `dlopen` in changes from the frontend to the backend. Do you have any results?

Reply:
I haven't done a super formal evaluation. What I can provide is the following:

Running: `python3 benchmark.py run -p grover -m runtime -i catalyst/lightning.qubit --nlayers 1000 -n 10` runs the grover benchmark as developed by Sergei with 1000 layers (which looking at the plots corresponds to around than 30000 gates) 10 times.

Running it with lightning.qubit gave me an average of 2.93625972270966 seconds and a standard deviation of 0.000616097273981.
Running it with a dummy device which was just a wrapper around lightning.qubit game me an average of 2.9408891916275 with a standard deviation of 0.000891513455482.

Again, I did not go for a t-test because the slowdown is 1.002x which in my opinion is not that much (1x slowdown is equal time). We could do a t-test but I don't think it is worth while.